cognition_objects/strategy_step.py

```diff
@@ -66,6 +66,7 @@ def create(
     progress_text: str,
     with_commit: bool = True,
     created_at: Optional[datetime] = None,
+    execute_if_source_code: Optional[str] = EXECUTE_IF_SOURCE_CODE,
```

Comment:
Not how Optional is supposed to be used & antipattern for default values as a change of execute_if_source_code would change the default value.
```suggestion
    execute_if_source_code: Optional[str] = None,
    ) -> CognitionStrategyStep:
    if not execute_if_source_code:
        execute_if_source_code = EXECUTE_IF_SOURCE_CODE
```